File: Src/Icm.Core/Search and Replace/PlainStringGenerator.vb

Namespace Icm.Text

    ''' <summary>
    ''' Returns always the string passed to the constructor.
    ''' </summary>
    ''' <remarks></remarks>
    Public Class PlainStringGenerator
        Implements IEnumerator(Of String)
        Implements IEnumerable(Of String)

        Private ReadOnly s_ As String

        Public Sub New(ByVal s As String)
            s_ = s
        End Sub

        Protected Sub New()
        End Sub

        Public ReadOnly Property Current As String Implements IEnumerator(Of String).Current
            Get
                Return s_
            End Get
        End Property

        Public ReadOnly Property Current1 As Object Implements IEnumerator.Current
            Get
                Return Current
            End Get
        End Property

        Public Function MoveNext() As Boolean Implements IEnumerator.MoveNext
            Return False
        End Function

        Public Sub Reset() Implements IEnumerator.Reset
        End Sub


        Public Sub Dispose() Implements IDisposable.Dispose
        End Sub

        Public Function GetEnumerator() As IEnumerator(Of String) Implements IEnumerable(Of String).GetEnumerator
            Return Me
        End Function

        Public Function GetEnumerator1() As IEnumerator Implements IEnumerable.GetEnumerator
            Return Me
        End Function
    End Class
End Namespace

File: Src/Icm.Core/Search and Replace/PlainStringGenerator.cs
using System.Collections;
using System.Collections.Generic;

namespace Icm.Text
{

	/// <summary>
	/// Returns always the string passed to the constructor.
	/// </summary>
	/// <remarks></remarks>
	public class PlainStringGenerator : IEnumerator<string>, IEnumerable<string>
	{


		private readonly string s_;
		public PlainStringGenerator(string s)
		{
			s_ = s;
		}

		protected PlainStringGenerator()
		{
		}

		public string Current {
			get { return s_; }
		}

		public object Current1 {
			get { return Current; }
		}
		object IEnumerator.Current {
			get { return Current1; }
		}

		public bool MoveNext()
		{
			return false;
		}

		public void Reset()
		{
		}


		public void Dispose()
		{
		}

		public IEnumerator<string> GetEnumerator()
		{
			return this;
		}

		public IEnumerator GetEnumerator1()
		{
			return this;
		}
		IEnumerator IEnumerable.GetEnumerator()
		{
			return GetEnumerator1();
		}
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
